includes/download.clip.inc

```diff
@@ -16,7 +16,13 @@ function islandora_openseadragon_download_clip(AbstractObject $object) {
     module_load_include('inc', 'islandora_openseadragon', 'includes/utilities');
     $clip_parts = islandora_openseadragon_construct_clip_url($_GET['clip'], TRUE);
     if ($clip_parts) {
-      $filename = $object->label;
+      $regex = '/[^a-zA-z0-9\s]/';
+      $space = '/\s/';
+      $filename = strtolower($object->label);
+      if (preg_match($regex, $filename) || preg_match($space, $filename)) {
```

Comment:
This is fine but feels like an overkill of regular expressions. Any chance you could simplify this? Like checking if something matches, before applying the replacement  is basically calling all too many tines

Reply:
something along these lines?
`      $regex = '/[^a-zA-z0-9\s]/';
      $space = '/\s/';
      $filename = strtolower($object->label);
      if (preg_match($space, $filename)) {
        $filename = preg_replace($space, '_', $escaped);
      }

      if (preg_match($regex, $filename)) {
        $filename = preg_replace($regex, '', $filename);
      } `

Reply:
https://github.com/IAMlKeno what about directly sanitizing instead of checking first? @nhart how do you feel about that? Look, D7 had an attempt to do the same. See 
https://www.drupal.org/project/drupal/issues/2472895
and in specific the implementation
https://www.drupal.org/files/issues/file_check_destination.patch

and search for `function file_check_name($name)`

Reply:
@DiegoPino any further input?